lib/super_diff/rspec/monkey_patches.rb

```diff
@@ -713,11 +713,9 @@ def actual_for_diff
 
           def expected_for_matcher_text
             if @expected_message
-              "#<#{@expected_error.name} #{description_of(@expected_message)}>"
-            elsif @expected_error.is_a? Regexp
-              "#<Exception #{description_of(@expected_error)}>"
+              "#<#{describe_expected_error} #{description_of(@expected_message)}>"
             else
-              "#<#{@expected_error.name}>"
+              "#<#{describe_expected_error}>"
             end
           end
```

Comment:
RSpec's RaiseError class, which represents the `raise_error` matcher, has an `expected_error` method that looks like this:

https://github.com/rspec/rspec-expectations/blob/35086f80e579d4c4e9cb2023bf34b4d278f4a436/lib/rspec/matchers/built_in/raise_error.rb#L222-L235

I tried plugging this and this is what I found:

- `raise_error(StandardError)` => `StandardError`
- `raise_error("some message")` => `Exception with "some message"`
- `raise_error(StandardError, "boo")` => `StandardError with "a"`
- `raise_error(a_kind_of(X)` => `Exception with #<a kind of X>`
- `raise_error(a_kind_of(Array).or eq(true))` => `Exception with #<a kind of Array or eq true>`
- `raise_error(double)` => `#<Double (anonymous)>`
- `raise_error(double, "boo")` => `#[Double (anonymous)] with "boo"`

I am not sure why I didn't adopt the original code more closely, but I think I was trying to make some of these representations look better.

It would be better, I think, if we could say:

- `raise_error(StandardError)` => `a StandardError`
- `raise_error("some message")` => `a RuntimeError with message "some message"`
- `raise_error(StandardError, "boo")` => `a StandardError with message "a"`
- `raise_error(a_kind_of(X)` => `an error matching #<a kind of X>`
- `raise_error(a_kind_of(Array).or eq(true))` => `an error matching #<a kind of Array or eq true>`
- `raise_error(double)` => `#<Double (anonymous)>`
- `raise_error(double, "boo")` => `#<Double (anonymous)> with message "boo"`

So for instance, we might see something like:

```
Expected raised exception #<Runtime "boo"> to match a StandardError
```

or

```
Expected raised exception #<Runtime "boo"> to match an error matching #<a kind of Array>
```

(That last example doesn't make much sense, but it's theoretically possible, so we ought to represent it somehow.)

I also realized that I'm not testing what happens when the "message" argument is a regexp. I didn't include it in the above list because of that reason, but we would have to account for that too.

All this to say — what do you think about the above changes to the described versions of the error/message? If you like them I'm happy to port your changes to a new branch and then make the appropriate modifications.

Reply:
Thinking about this a bit more, perhaps tweaking the error message is out of scope of this PR — so I will accept this PR given that it does solve the issue that you pointed out earlier!